sal-service/src/main/java/org/ow2/proactive/sal/service/service/NodeService.java

```diff
@@ -277,6 +277,23 @@ public List<Deployment> getNodes(String sessionId) throws NotConnectedException
         return allDeployments;
     }
 
+    /**
+     * Get all added nodes with specific node names
+     * @param sessionId A valid session id
+     * @param nodeNames Valid node names
+     * @return List of all table Deployment's entries
+     */
+    public List<Deployment> getNodesByNames(String sessionId, List<String> nodeNames) throws NotConnectedException {
```

Comment:
I believe it more redundant and elegant to have one getNodes function and have the if condition on the nodeNames inside of the getNodes function because the difference is very minimal: 
 "repositoryService.listDeployments();" or "repositoryService.findAllDeployments(nodeNames);"

Reply:
In terms of code I can agree with you. But in the long term scenario the service classes are the ones that will be handling more complexity, so I preferred here to separate methods based on their responsabilities to make Service classes (and thus their test classes) easier to maintain.